gr-vocoder/lib/freedv_tx_ss_impl.cc

```diff
@@ -70,7 +70,8 @@ freedv_tx_ss_impl::freedv_tx_ss_impl(int mode,
     if ((d_freedv = freedv_open(mode)) == NULL)
         throw std::runtime_error("freedv_tx_ss_impl: freedv_open failed");
 #endif
-    snprintf(d_cb_state.tx_str, 79, "%s", d_msg_text.c_str());
+    std::string txstr = str(boost::format("%s") % d_msg_text.c_str());
```

Comment:
One of these is not like the other: The left hand has the 79-char limit, the right hand does not, and `txstr` serves no purpose, nor does `boost::format()`.

Reply:
Why not replace with `auto txstr = d_msg_text.substr(0,79)`?

But also, what's so bad about `snprintf()`? Seems like a safe way to cap a string and always have it be null-terminated.

Reply:
In terms of "code smell", `strcpy` is much more pungent than `snprintf`.